src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -13738,6 +13815,14 @@ int Assembler::simd_prefix_and_encode(XMMRegister dst, XMMRegister nds, XMMRegis
   }
 }
 
+bool Assembler::is_demotable(bool no_flags, int dst_enc, int nds_enc, int src_enc) {
+  return (!no_flags && dst_enc == nds_enc);
+}
+
+bool Assembler::is_demotable(bool no_flags, int dst_enc, int nds_enc) {
+  return (!no_flags && dst_enc == nds_enc);
+}
+
```

Comment:
Just to let you know, we are missing a case where dst_enc can be equal to src_enc; in that case, we can still demote EVEX to REX/REX2 encoding along with a change in primary opcode if needed.

This will apply to all the commutative operations (ADD/ AND / OR / XOR)

Reply:
Thanks for the proposal. This feature will be enabled in another PR using the JBS entry https://bugs.openjdk.org/browse/JDK-8354348.